StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/ViewControllers/UpdatePaymentMethodViewController.swift

```diff
@@ -0,0 +1,378 @@
+                                        appearance: appearance)
+        navBar.delegate = self
+        navBar.setStyle(navigationBarStyle())
+        return navBar
+    }()
+
+    // MARK: Views
+    lazy var formStackView: UIStackView = {
+        let stackView = UIStackView(arrangedSubviews: [headerLabel, paymentMethodDetails, updateButton, removeButton, errorLabel])
+        stackView.isLayoutMarginsRelativeArrangement = true
+        stackView.axis = .vertical
+        stackView.setCustomSpacing(16, after: headerLabel) // custom spacing from figma
+        stackView.setCustomSpacing(32, after: paymentMethodDetails) // custom spacing from figma
+        stackView.setCustomSpacing(16, after: updateButton) // custom spacing from figma
+        return stackView
+    }()
+
+    private lazy var headerLabel: UILabel = {
+        let label = PaymentSheetUI.makeHeaderLabel(appearance: appearance)
+        label.text = viewModel.header
+        return label
+    }()
+
+    private lazy var updateButton: ConfirmButton = {
+        let button = ConfirmButton(state: .disabled, callToAction: .custom(title: .Localized.save), appearance: appearance, didTap: {  [weak self] in
+            switch self?.viewModel.paymentMethod.type {
+            case .card:
+                Task {
+                    await self?.updateCard()
+                }
+            default:
+                fatalError("Updating payment method has not been implemented for \(self?.viewModel.paymentMethod.type ?? .unknown)")
+            }
+        })
+        button.isHidden = !viewModel.canEdit
+        return button
+    }()
+
+    private lazy var removeButton: UIButton = {
+        let button = UIButton(type: .custom)
+        if #available(iOS 15.0, *) {
+            var configuration = UIButton.Configuration.bordered()
+            configuration.contentInsets = NSDirectionalEdgeInsets(top: 10, leading: 16, bottom: 10, trailing: 16)
+            configuration.baseBackgroundColor = .clear
+            button.configuration = configuration
+        } else {
+            button.contentEdgeInsets = UIEdgeInsets(top: 10, left: 16, bottom: 10, right: 16)
+        }
+        button.setTitleColor(appearance.colors.danger, for: .normal)
+        button.layer.borderColor = appearance.colors.danger.cgColor
+        button.layer.borderWidth = appearance.primaryButton.borderWidth
+        button.layer.cornerRadius = appearance.cornerRadius
+        button.setTitle(.Localized.remove, for: .normal)
+        button.titleLabel?.textAlignment = .center
+        button.titleLabel?.font = appearance.scaledFont(for: appearance.font.base.medium, style: .callout, maximumPointSize: 25)
+        button.titleLabel?.adjustsFontForContentSizeCategory = true
+        button.addTarget(self, action: #selector(removePaymentMethod), for: .touchUpInside)
+        button.isHidden = !viewModel.canRemove
+        return button
+    }()
+
+    private lazy var paymentMethodDetails: UIStackView = {
```

Comment:
Nit: `paymentMethodDetails` makes this sound like it's an object contains data for a payment method. Maybe rename this to `paymentMethodForm`

Reply:
Done